src/main/scala/AssemblyHelpers.scala

```diff
@@ -10,7 +10,15 @@ object AssemblyHelpers {
 
   val settings: Seq[Setting[_]] =
     Seq(
-      assembly := ReproducibleBuildsPlugin.postProcessJar(assembly.value)
+      assembly := {
+        if ((assemblyOption in assembly).value.prependShellScript.isDefined) {
+          val log = streams.value.log
+          log.warn("Cannot make assembly reproducible when prependShellScript is set")
+          assembly.value
+        }
+        else {
+          ReproducibleBuildsPlugin.postProcessJar(assembly.value)
+        }
```

Comment:
This doesn't pass the 'scripted' tests - I think you need to move the `assembly.value` and `streams.value` outside of the `if` to make sbt understand it.